From 048536589e792a630c542da87c49427bcb8354de Mon Sep 17 00:00:00 2001
From: Sudarshan S <sudarshan.s@intel.com>
Date: Sun, 7 Feb 2021 18:21:32 +0530
Subject: [PATCH 4/4] vaDeriveImage: Enable WaDisableGmmLibOffsetInDeriveImage
 for JSL/EHL

Enables the software workaround to disable calculation of the UV
offset by gmmlib. An incorrect UV offset was by returned  gmmlib
in vaDeriveImage which casues the shift in the color planes(UV).
This was observed during a chromecast session in mirror mode on ChromeOS
with H264 encode.

(cherry picked from commit 3d0212e9ab4b4e4a5414152ad8fb92f996baa08a)
---
 media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp b/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp
index e336cb37..bf231d6b 100644
--- a/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp
+++ b/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp
@@ -413,6 +413,9 @@ static bool InitEhlMediaWa(struct GfxDeviceInfo *devInfo,
     MEDIA_WR_WA(waTable, Wa16KInputHeightNV12Planar420, 1);
     MEDIA_WR_WA(waTable, WaDisableCodecMmc, 1);
 
+    /*Software workaround to disable the UV offset calculation by gmmlib
+      CPU blt call will add/remove padding on the platform*/
+    MEDIA_WR_WA(waTable, WaDisableGmmLibOffsetInDeriveImage, 1);
     return true;
 }
 
-- 
2.17.1

